src/webgl/light.js

```diff
@@ -39,32 +39,16 @@ var p5 = require('../core/core');
  *
  */
 p5.prototype.ambientLight = function(v1, v2, v3, a){
-  var gl = this._renderer.GL;
-  var shaderProgram = this._renderer._getShader(
-    'lightVert', 'lightTextureFrag');
-
-  gl.useProgram(shaderProgram);
-  shaderProgram.uAmbientColor = gl.getUniformLocation(
-    shaderProgram,
-    'uAmbientColor[' + this._renderer.ambientLightCount + ']');
-
   var color = this._renderer._pInst.color.apply(
     this._renderer._pInst, arguments);
-  var colors = color._array;
-
-  gl.uniform3f( shaderProgram.uAmbientColor,
-    colors[0], colors[1], colors[2]);
+  var colors = color._array.slice(0, 3);
 
-  //in case there's no material color for the geometry
-  shaderProgram.uMaterialColor = gl.getUniformLocation(
-    shaderProgram, 'uMaterialColor' );
-  gl.uniform4f( shaderProgram.uMaterialColor, 1, 1, 1, 1);
+  this._renderer._setUniform('uAmbientColor[' +
```

Comment:
I didn't want to assume that the shader program necessarily exists yet when this function is called, so instead I went with a system where uniform data is saved first, and then applied to the shader program at render-time.

Right now, we've got a distinction between renderer._setUniform, which can set the uniform for all shader programs, and shader.setUniform, which only sets it for a specific program. I'd like to keep this distinction, but think that there's probably a way to make it more understandable, either through different function naming, or better documentation, or maybe moving all global uniform setting to the p5.Shader class. Any thoughts?